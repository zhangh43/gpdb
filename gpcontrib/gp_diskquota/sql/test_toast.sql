-- Test toast
CREATE SCHEMA s5;
SELECT diskquota.set_schema_quota('s5', '1 MB');
SET search_path TO s5;
CREATE TABLE a5 (message text);
INSERT INTO a5
SELECT (SELECT 
        string_agg(chr(floor(random() * 26)::int + 65), '')
        FROM generate_series(1,10000)) 
FROM generate_series(1,10);

SELECT pg_sleep(5);
-- expect insert toast fail
INSERT INTO a5
SELECT (SELECT 
        string_agg(chr(floor(random() * 26)::int + 65), '')
        FROM generate_series(1,100000)) 
FROM generate_series(1,1000000);

DROP TABLE a5;
RESET search_path;
DROP SCHEMA s5;

