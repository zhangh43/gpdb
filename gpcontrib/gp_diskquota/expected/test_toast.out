-- Test toast
CREATE SCHEMA s5;
SELECT diskquota.set_schema_quota('s5', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SET search_path TO s5;
CREATE TABLE a5 (message text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'message' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO a5
SELECT (SELECT 
        string_agg(chr(floor(random() * 26)::int + 65), '')
        FROM generate_series(1,10000)) 
FROM generate_series(1,10);
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect insert toast fail
INSERT INTO a5
SELECT (SELECT 
        string_agg(chr(floor(random() * 26)::int + 65), '')
        FROM generate_series(1,100000)) 
FROM generate_series(1,1000000);
ERROR:  schema's disk space quota exceeded with name:s5
DROP TABLE a5;
RESET search_path;
DROP SCHEMA s5;
